src/pipeline.py

```diff
@@ -31,32 +38,54 @@ def run(self):
         while True:
             work = work_queue.get()
             if work is None:
+                print("done")
                 return
             print(f"Received work {work}")
+            executor = work.executor
+            item = work.item
+            executor.run(item)
+            print("Done processing work")
+            if executor.next is not None:
+                work_queue.put(Work(executor.next, item))
+            work_queue.task_done()
 
-def main():
-    parser = PipelineCLIParser(description='AutoDash pipeline for processing (near-)accident dashcam footage')
-    args = parser.parse_args()
 
-    hostname = gethostname()
+def run(executors, **kwargs):
+    work_queue = JoinableQueue()
+
     context = {
-        **vars(args),
-        'work_queue': Queue(),
-        'hostname': hostname
+        **kwargs,
+        'work_queue': work_queue,
     }
 
     workers = [
-        PipelineWorker(context, name=f"worker-{hostname}-{i}")
+        PipelineWorker(context, name=f"worker-{i}")
         for i in range(0, context['n_workers'])
     ]
 
     # Start all workers
     for worker in workers:
         worker.start()
 
+    iterations = context['max_iterations']
+    iterations = 10000 if iterations is None else iterations
+
+    for i in range(iterations):
+        idx = 0
+        work_queue.put(Work(executors[idx], None))
+        idx += 1
```

Comment:
This puts the same executor on the work queue
```suggestion
    for executor in executors:
        work_queue.put(Work(executor, None))
```

Reply:
ya that's my bad, what I meant to do was alternate between all of the ones in the list, iterations times (that way the code can end for testing at least)